backend/frontend/tests/test_progress_details_pages.py

```diff
@@ -307,3 +310,64 @@ def test_progress_details_displays_correct_details_for_topic(
             self.topic.content_url,
             body,
         )
+
+
+class TestRepoProjectDetailsPage(FrontendTestMixin):
+    def setUp(self):
+        super().setUp()
+        self.user = UserFactory(email="learner_1@umuzi.org")
+        self.user.set_password(self.user.email)
+        self.user.save()
+        self.do_login(self.user)
+
+    def _get_project_progress_details_url(self):
+        return self.reverse_url(
+            PROGRESS_DETAILS_VIEW,
+            kwargs={
+                "content_type": "project",
+                "id": self.card.recruit_project.id,
+            },
+        )
+
+    def make_ip_project_card(self):
+        self.card = AgileCardFactory()
+        self.card.recruit_project.repository = RepositoryFactory(user=self.user)
+        self.card.recruit_project.save()
+        self.card.save()
+
+        self.card.assignees.set([self.user])
+        self.card.recruit_project.recruit_users.set([self.user])
+
+    def test_progress_details_page_displays_repo_for_repo_project(self):
+        self.make_ip_project_card()
+
+        repo_project_progress_details_url = self._get_project_progress_details_url()
+
+        self.page.goto(repo_project_progress_details_url)
+
+        self.page.wait_for_load_state()
+
+        repo_link = self.page.locator("a#repo-link")
+
+        expect(repo_link).to_contain_text(
+            self.card.recruit_project.repository.full_name
+        )
+
+    def test_progress_details_page_displays_open_prs_for_repo_project(self):
+        self.make_ip_project_card()
+
+        pr = PullRequestFactory(
+            repository=self.card.recruit_project.repository, state=PullRequest.OPEN
+        )
+
+        repo_project_progress_details_url = self._get_project_progress_details_url()
+
+        self.page.goto(repo_project_progress_details_url)
+
+        self.page.wait_for_load_state()
+
+        pr_link = self.page.locator(f"a#pr_{pr.number}")
+
+        expect(pr_link).to_contain_text(
```

Comment:
I'm not sure if it's possible to make this test work without playwright? Eg just generate HTML and look at it without actually spinning up a browser?